### PR TITLE
[soem]Update version to latest.

### LINKED ISSUE #707: Release version bump
Hi,

are there any plans to release a new version of this library?

### PR DESCRIPTION
**Pull request by omarhogni**
Fixes https://github.com/OpenEtherCATsociety/SOEM/issues/707

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/soem/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "soem",
-  "version": "1.4.0",
-  "port-version": 4,
+  "version-date": "2022-12-02",
```
Comment (FrankXie05) [inline ports/soem/vcpkg.json:3]:
```suggestion
  "version-date": "2023-06-09",
```
The data here refers to the date when vcpkg updated this port to ensure when we have modified it.

File: ports/soem/portfile.cmake [hunk 2]
```diff
@@ -3,12 +3,9 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
-    SHA512 2967775c6746bb63becea5eb12f136c184bbf874e1e5e8753374bfc212ec9cefbf1159350e79627b978af3562d261b61c50f38936a425c4d9c70598a1d136817
+    REF a901500618405760a564e64a6816705e29f50f9f
+    SHA512 d554bc1c3780b1a81402a7fda490f516caba6bd943a28482740b5c9d97e4273a11546e79c92796487ee9901f568cbf1b329d4e1c1d32602fdce0088a77c82443
     HEAD_REF master
-    PATCHES
-        winpcap.patch
-        disable-werror.patch
 )
 
 vcpkg_cmake_configure(
```
Comment (FrankXie05) [inline ports/soem/portfile.cmake:11]: We don't build `TEST`. Please add option `BUILD_TESTS=OFF`.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/soem/portfile.cmake [hunk 3]
```diff
@@ -3,12 +3,9 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO OpenEtherCATsociety/SOEM
-    REF abbf0d42e38d6cfbaa4c1e9e8e07ace651c386fd #v1.4.0
-    SHA512 2967775c6746bb63becea5eb12f136c184bbf874e1e5e8753374bfc212ec9cefbf1159350e79627b978af3562d261b61c50f38936a425c4d9c70598a1d136817
+    REF a901500618405760a564e64a6816705e29f50f9f
+    SHA512 d554bc1c3780b1a81402a7fda490f516caba6bd943a28482740b5c9d97e4273a11546e79c92796487ee9901f568cbf1b329d4e1c1d32602fdce0088a77c82443
     HEAD_REF master
-    PATCHES
-        winpcap.patch
-        disable-werror.patch
```
Comment (FrankXie05) [inline ports/soem/portfile.cmake:11]: Upstream has been fixed the patches.

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/soem/portfile.cmake (hunk 3 above)
Comment (omarhogni) [reply ports/soem/portfile.cmake]: I misunderstood the winpcap patch and have added it again.

Regarding the werror patch it looks like it was added when CI pipelines for vcpkg were updated to ubuntu 20.04 and look to no longer be required.

**Comment by dg0yt** [comment]
> Regarding the werror patch it looks like it was added when CI pipelines for vcpkg were updated to ubuntu 20.04 and look to no longer be required.

Please restore the patch. werror is a tool for a particular build environment of upstream developers, but not for general build recipes which may face different (incl. future) environments, facing unexpected warnings incl. false positives.

**Comment by omarhogni** [comment]
> > Regarding the werror patch it looks like it was added when CI pipelines for vcpkg were updated to ubuntu 20.04 and look to no longer be required.
> 
> Please restore the patch. werror is a tool for a particular build environment of upstream developers, but not for general build recipes which may face different (incl. future) environments, facing unexpected warnings incl. false positives.

Added back werror patch.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**